server/app.js

```diff
@@ -0,0 +1,250 @@
+ */
+const swaggerHandler = (newInstanceMethods, method, ...args) => {
+  let middlewareArgs = [...args];
+  const [firstArg] = middlewareArgs;
+  const [lastArg] = middlewareArgs.slice(-1);
+  // The first argument is a string and does not equal the last argument and therefore
+  // this is a path definition and it should be added to the app's schema
+  if (typeof firstArg === 'string' && lastArg !== firstArg) {
+    if (typeof lastArg === 'function') {
+      // This means the options object was not given and therefore the
+      // last argument should be considered as the controller for the given path
+      handleLastArg(lastArg, firstArg, method);
+    } else if (typeof lastArg === 'object') {
+      // This means the options object was given and therefore the second last argument should be
+      // considered as the controller for the given path. This options argument is
+      // therefore removed when sending it to the newInstanceMethods[method]
+      // because the express application expects a function
+      const [secondLastArg] = middlewareArgs.slice(-2);
+      middlewareArgs = middlewareArgs.slice(0, middlewareArgs.length - 1);
+      handleLastArg(secondLastArg, firstArg, method, lastArg);
+    }
+  }
+  // This needs to return the output of the bound method's function
+  // Do not return before this stage
+  return newInstanceMethods[method](...middlewareArgs);
+};
+
+/**
+ * Replaces the methods of the given instance with that which will allow us to capture the route
+ * definition and document it with swagger
+ * The app methods being replaced include app.all, app.use, app.[HTTP method] e.g, app.get
+ * @param {express} appInstance The express application instance whose methods need to be replaced
+ * @returns {Promise<void>} A promise that resolves after the methods
+ * have been successfully replaced */
+const replaceMethods = async (appInstance) => {
+  // The new instance methods that point to the original function definitions for the app instance
+  // Here we bind the appInstance, so that "this" refers to the
+  // appInstance when the method is called
+  const newInstanceMethods = {
+    use: appInstance.use.bind(appInstance),
+    all: appInstance.all.bind(appInstance),
+  };
+
+  // Loop one by one adding the original function definitions to the newInstanceMethods object
+  // only if the appInstance exposes the said HTTP method
+  await loopSequentially(httpMethods, (method) => {
+    newInstanceMethods[method] = appInstance[method] && appInstance[method].bind(appInstance);
+  });
+
+  // Loop one by one replacing the appInstance's methods found in the keys of the
+  // newInstanceMethods with our own function that passes the arguments to our swagger
+  // handler which will then handle the addition of the paths, if any, to the swagger schema
+  await loopSequentially(Object.keys(newInstanceMethods), (method) => {
+    if (method) {
```

Comment:
@davenjeru when would no method be provided here?

Reply:
The 'View' method is given by postman as a HTTP method but there is no method like `app.view` for an express app.